Content.Server/Chat/SuicideSystem.cs

```diff
@@ -123,19 +135,52 @@ private bool EnvironmentSuicideHandler(EntityUid victim, SuicideEvent suicideEve
             return false;
         }
 
-        private void ApplyDeath(EntityUid target, SuicideKind kind)
+        private void ApplyLethalDamage(EntityUid target, DamageSpecifier? damage)
         {
-            if (kind == SuicideKind.Special)
+            if (!TryComp<DamageableComponent>(target, out var damagable) || !TryComp<MobThresholdsComponent>(target, out var thresholds))
                 return;
 
-            if (!_prototypeManager.TryIndex<DamageTypePrototype>(kind.ToString(), out var damagePrototype))
+            var lethalAmountOfDamage = thresholds.Thresholds.Keys.Last() - damagable.TotalDamage;
+
+            if (damage == null)
+            {
+                var damagePrototype = _prototypeManager.Index<DamageTypePrototype>("Blunt");
+                damage = new DamageSpecifier(damagePrototype, lethalAmountOfDamage);
+            }
+
+            var finalDamage = new DamageSpecifier(damage);
+            finalDamage.DamageDict.Remove("Structural");
+            var totalItemDamage = finalDamage.GetTotal();
+            foreach (var (key, value) in finalDamage.DamageDict)
+            {
+                finalDamage.DamageDict[key] = Math.Ceiling((double) (value * lethalAmountOfDamage / totalItemDamage));
+            }
+            _damageableSystem.TryChangeDamage(target, finalDamage, true, origin: target);
+        }
+
+        private void ApplyLethalDamage(EntityUid target, ProtoId<DamageTypePrototype>? kind)
```

Comment:
Add a summary comment indicating that this function is for applying lethal damage in a single damage type